src/libraries/System.Memory/tests/ReadOnlySpan/Count.byte.cs

```diff
@@ -0,0 +1,313 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Numerics;
+using Xunit;
+
+namespace System.SpanTests
+{
+    public static partial class ReadOnlySpanTests
+    {
+        [Fact]
+        public static void ZeroLengthCount_Byte()
+        {
+            ReadOnlySpan<byte> span = new ReadOnlySpan<byte>(Array.Empty<byte>());
+
+            int count = span.Count<byte>(0);
+            Assert.Equal(0, count);
+        }
+        
+        [Fact]
+        public static void ZeroLengthCount_RosByte()
+        {
+            ReadOnlySpan<byte> span = new ReadOnlySpan<byte>(Array.Empty<byte>());
+
+            int count = span.Count<byte>(new ReadOnlySpan<byte>(new byte[] { 0, 0 }));
+            Assert.Equal(0, count);
+        }
+        
+        [Fact]
+        public static void DefaultFilledCount_Byte()
+        {
+            for (int length = 0; length <= byte.MaxValue; length++)
+            {
+                byte[] a = new byte[length];
+                ReadOnlySpan<byte> span = new ReadOnlySpan<byte>(a);
+
+                for (int i = 0; i < length; i++)
+                {
+                    byte target0 = default;
+
+                    int count = span.Count(target0);
+                    Assert.Equal(length,  count);
+                }
+            }
+        }
+        
+        [Fact]
+        public static void DefaultFilledCount_RosByte()
+        {
+            for (int length = 0; length <= byte.MaxValue; length++)
+            {
+                byte[] a = new byte[length];
+                ReadOnlySpan<byte> span = new ReadOnlySpan<byte>(a);
+
+                for (int i = 0; i < length; i++)
```

Comment:
Same question; this inner loop doesn't look necessary... nothing in the body refers to i, so it's just running the same test repeatedly.